utilities/tools/Register-AzureDevOpsPipelines.ps1

```diff
@@ -0,0 +1,216 @@
+
+.PARAMETER RelativePipelinePath
+Optional. The relative local path to the folder with the pipeline YAML files. Make sure your workspace is opened in the repository root.
+Defaults to '.azuredevops/modulePipelines'.
+
+.PARAMETER CreateBuildValidation
+Optional. Create an additional pull request build validation rule for the pipelines.
+
+.EXAMPLE
+$inputObject = @{
+    OrganizationName      = 'Contoso'
+    ProjectName           = 'CICD'
+    SourceRepository      = 'Azure/ResourceModules'
+    GitHubPAT             = '<Placeholder>'
+    AzureDevOpsPAT        = '<Placeholder>'
+}
+Register-AzureDevOpsPipelines @inputObject
+
+Registers all pipelines in the default path in the DevOps project [Contoso/CICD] by leveraging the given AzureDevOpsPAT and creating a service connection to GitHub using the provided GitHubPAT
+
+.NOTES
+You'll need the 'azure-devops' extension to run this function: `az extension add --upgrade -n azure-devops`
+
+The steps you'd want to follow are
+- (if pipelines are in GitHub) Create a service connection to the target GitHub repository using e.g. oAuth
+- Create a PAT token for the Azure DevOps environment in which you want to register the pipelines in
+- Run this script with the corresponding input parameters
+- Create any required element required to execute the pipelines. For example:
+  - Library group(s) used in the pipeline(s)
+  - Service connection(s) used in the pipeline(s)
+  - Agent pool(s) used in the pipeline(s) if not using the default available agents
+#>
+function Register-AzureDevOpsPipelines {
+
+    [CmdletBinding(SupportsShouldProcess)]
+    param (
+        [Parameter(Mandatory = $true)]
+        [string] $OrganizationName,
+
+        [Parameter(Mandatory = $true)]
+        [string] $ProjectName,
+
+        [Parameter(Mandatory = $true)]
+        [string] $AzureDevOpsPAT,
+
+        [Parameter(Mandatory = $false)]
+        [string] $SourceRepository = 'Azure/ResourceModules',
+
+        [Parameter(Mandatory = $false)]
+        [ValidateSet('gitHub', 'tfsgit')]
+        [string] $SourceRepositoryType = 'gitHub',
+
+        [Parameter(Mandatory = $false)]
+        [string] $GitHubPAT,
+
+        [Parameter(Mandatory = $false)]
+        [string] $GitHubServiceConnectionName = $SourceRepository,
+
+        [Parameter(Mandatory = $false)]
+        [string] $BranchName = 'main',
+
+        [Parameter(Mandatory = $false)]
+        [string] $PipelineTargetPath = 'Modules',
+
+        [Parameter(Mandatory = $false)]
+        [string] $RelativePipelinePath = '.azuredevops/modulePipelines',
+
+        [Parameter(Mandatory = $false)]
+        [bool] $CreateBuildValidation = $false
+    )
+
+    Write-Verbose '##############'
+    Write-Verbose '# Local Data #'
+    Write-Verbose '##############'
+    Write-Verbose 'Identify relevant Azure Pipelines to be updated'
+    # $RelativePipelinePath = Join-Path (Get-Location).Path $RelativePipelinePath
+    $RelativePipelinePath = $RelativePipelinePath
```

Comment:
This might be redundant code?

Reply:
removed

Reply:
$RelativePipelinePath = $RelativePipelinePath

It is still there ? :D

Reply:
And I only read the comment above it :D

Reply:
fixed